tdp/core/collections.py

```diff
@@ -195,6 +196,19 @@ def _init_operations(self):
                     collection_name=collection_name,
                 )
 
+    def get_services(self) -> set[str]:
+        """Get the list of services names."""
+        return {operation.service_name for operation in self.operations.values()}
+
+    def get_default_vars_path(
+        self, service_name: str
+    ) -> Optional[Generator[Path, None, None]]:
+        """Get the default vars paths for a service."""
+        for collection in self._collections.values():
+            default_vars = collection.get_service_default_vars(service_name)
+            if default_vars:
+                return default_vars
```

Comment:
Maybe you want to `yield` here instead of `return` ?

Reply:
`Collection.get_service_default_vars` returns a Generator.

Replacing `return` by `yield` would lead to a `Generator[Generator[Path, None, None], None, None]`, I'm not sure that what we want to obtain.

Reply:
Indeed but you stop the loop over collections at the first collection where there are defaults vars, a service can have default vars in multiple collection.

Reply:
I fixed it!